src/libcore/thread.cpp

```diff
@@ -85,12 +91,30 @@ class WorkerThread : public Thread {
 
 std::atomic<uint32_t> WorkerThread::m_counter{0};
 
+struct ThreadRegistrator {
```

Comment:
Registrator sounds weird. Perhaps "Notifier"?

Reply:
Actually, can we not just get rid of this class completely now?

Reply:
`static thread_local ThreadRegistrator;` will register all the external threads. I still we still want that, no?